src/main/java/controller/NameInputController.java

```diff
@@ -0,0 +1,118 @@
+package controller;
+
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.TextField;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.KeyEvent;
+import javafx.scene.layout.GridPane;
+import javafx.scene.text.Text;
+import javafx.stage.Stage;
+import utility.Settings;
+
+import java.io.IOException;
+import java.net.URL;
+
+/**
+ * Input for player name.
+ */
+public class NameInputController extends GridPane {
+
+    @FXML private Text nameInputPrompt;
+    @FXML private TextField nameInputField;
+    @FXML private Button cancelButton;
+    @FXML private Button enterButton;
+
+    private int number;
+    private String playerName;
+
+    /**
+     * A GridPane which asks for the name of the player.
+     * @param playerNumber Number of the current player.
+     */
+    public NameInputController(int playerNumber) {
+        this.number = playerNumber;
+        this.playerName = String.format("P%d", playerNumber);
+        loadFxml();
+        setText();
+    }
+
+    private void loadFxml() {
+        URL url = this.getClass().getClassLoader().getResource("playerNameInput.fxml");
+        FXMLLoader loader = new FXMLLoader(url);
+        loader.setRoot(this);
+        loader.setController(this);
+        try {
+            loader.load();
+        } catch (IOException exception) {
+            throw new RuntimeException(exception);
+        }
+    }
+
+    private void setText() {
+        nameInputPrompt.setText("Enter name " + this.playerName);
+        nameInputField.setPromptText(this.playerName);
+    }
+
+    /**
+     * Function to continue when the enter is pressed.
+     * @param event KeyEvent that is fired.
+     */
+    @FXML
+    public void handleEnterPressed(KeyEvent event) {
+        if (event.getCode() == KeyCode.ENTER) {
+            enterNameInput();
+        }
+    }
+
+    /**
+     * Function for what happens when the enter button is pressed.
+     */
+    @FXML
+    public void enterNameInput() {
```

Comment:
Is it an idea to force a placer to type it's name? Now one doesn't necessarily has to input a name, since otherwise P1 or P2 is used. It's not a bug, but maybe an improvement?
<feature